src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.ts

```diff
@@ -110,31 +127,38 @@ export class MongoDBEventStore implements EventStore<number> {
   async appendToStream<EventType extends Event>(
     streamName: StreamName,
     events: EventType[],
-    options?: AppendToStreamOptions<number> & {
-      /**
-       * These will be ran after a the events have been successfully appended to
-       * the stream. `appendToStream` will return after every projection is completed.
-       */
-      projections?: Array<Projection<EventType>>;
-    },
-  ): Promise<AppendToStreamResult<number>> {
-    let stream = await this.collection.findOne({
-      streamName: { $eq: streamName },
-    });
-    let currentStreamPosition = stream?.events.length ?? 0;
+    options?: AppendToStreamOptions,
+  ): Promise<AppendToStreamResult> {
+    let stream = streamPositionDeserializer(
```

Comment:
**FIX:** This could be made in a another PR, but when we're appending the stream, then we should not be reading it, we can use retries and checks on the expected version to ensure that we're not overriding the stream in the unexpected state.

Reply:
How would you recommend setting the `streamPosition` at the event metadata level when we go to append the events? Or is it not possible to store that without reading the stream first, and can just be encapsulated by the index in the array?

Reply:
@alex-laycalvert Yes, it's possible; if you don't provide append options or choose to use no concurrency check, then the event should be just appended and the version auto-incremented. If autoincrement doesn't work for bigints in MongoDB, then we can use number for stream position.

Reply:
Does MongoDB natively support an auto incrementing field or would this be in the form of a separate document as a `counter`?

Reply:
It does, by using $inc operator https://www.mongodb.com/docs/manual/reference/operator/update/inc/.

At least, that’s how I did it so far.